src/cmdstan/command.hpp

```diff
@@ -264,16 +265,12 @@ int command(int argc, const char *argv[]) {
           save_iterations, refresh, interrupt, logger, init_writer,
           sample_writers[0], diagnostic_json_writers[0]);
     } else {
-      std::string output_file
-          = get_arg_val<string_argument>(parser, "output", "file");
-      auto base_sfx = get_basename_suffix(output_file);
-      if (base_sfx.second.empty())
-        base_sfx.second = ".csv";
-      auto ofs = std::make_unique<std::ofstream>(base_sfx.first + "_pathfinder"
-                                                 + base_sfx.second);
-      if (sig_figs > -1) {
+      std::vector<std::string> output_filenames = make_filenames(
+          get_arg_val<string_argument>(parser, "output", "file"), "", ".csv", 1,
+          id);
+      auto ofs = std::make_unique<std::ofstream>(output_filenames[0]);
```

Comment:
Does this need the extra indirection with the helper function call and the vector return?